client/components/layout/channel_autocomplete.coffee

```diff
@@ -0,0 +1,81 @@
+{ defer } = require 'underscore'
+Modal = require 'simple-modal'
+sd = require('sharify').data
+async = require 'async'
+request = require 'superagent'
+User = require '../../models/user.coffee'
+
+module.exports.init = ->
+  $('#layout-sidebar-switch-channel').click =>
```

Comment:
This one big `init` function works fine for now—but I think ideally this gets refactored into a Backbone view for the sake of testability/modularity.